### ISSUE TITLE
Do we need models that consist of test data?

### ISSUE THREAD

**Issue by PackRuble** [PackRuble/weather_pack#2]
I see several use cases like this:
1. We don't have apiKey OWM, but want to test this package.
2. We can't see some data, such as weather events - `WeatherAlert`, which are not always available in the server response.
3. We want to provide initial data in our app. (or we afraid null, hehe)

We should to have abstract method to get weather.

**Comment by PackRuble** [comment]
It would be great if an option could be implemented to download `data_mock.json` to the user's local folder for further testing. 

On the other hand, why would we bother with this so much when we can just add a logger that would provide a url link of the current request. Then the user can go to the browser and download the actual answer themselves :) and dispose of it as they wish.

**Comment by PackRuble** [comment]
Implemented in version v1.1.0

For `GeocodingService` and `WeatherService` you can create a custom `OWMBuilder` for debugging and logging cases

```dart
class OWMBuilderCustom extends OWMBuilder {
  /// We output the url to the console for debugging and logging
  @override
  Future<T> getData<T>(
      {required Uri uri, required T Function(dynamic data) builder}) {
    print(uri);
    return super.getData(uri: uri, builder: builder);
  }
}

void workOwmBuilder({
  String api = 'your_apikey',
}) async {
  final customOWMBuilder = OWMBuilderCustom();
  final gService = GeocodingService(api, owmBuilder: customOWMBuilder);

  final List<PlaceGeocode> places = await gService.getLocationByCoordinates(
      latitude: 52.374, longitude: 4.88969);

  print(places);
}
```

or

For `OWMTestService` you can create a custom `Client`. A more low-level way, would require an explicit dependency on the `http` package:

```dart
class CustomClient extends IOClient {
  /// We output the url to the console
  @override
  Future<Response> get(Uri url, {Map<String, String>? headers}) {
    print(url);
    return super.get(url, headers: headers);
  }
}

/// We output the url to the console for debugging
void workCustomClient({
  String api = 'your_apikey',
}) async {
  final customClient = CustomClient();
  final testService = OWMTestService(api, customClient);

  final bool isValidKey = await testService.isValidApikey();

  print(isValidKey);
}
```
